src/adapter/src/coord/sequencer/inner/cluster.rs

```diff
@@ -214,18 +232,45 @@ impl Coordinator {
 
         match (&config.variant, new_config.variant) {
             (Managed(config), Managed(new_config)) => {
-                self.sequence_alter_cluster_managed_to_managed(
-                    Some(session),
-                    cluster_id,
-                    config,
-                    new_config,
-                    ReplicaCreateDropReason::Manual,
-                )
-                .await?;
+                let needs_finalization = self
+                    .sequence_alter_cluster_managed_to_managed(
+                        Some(session),
+                        cluster_id,
+                        config,
+                        new_config.clone(),
+                        ReplicaCreateDropReason::Manual,
+                        strategy.clone(),
+                    )
+                    .await?;
+                if needs_finalization {
+                    return match strategy.condition {
+                        AlterClusterStrategyCondition::None => Ok(StageResult::Response(
+                            ExecuteResponse::AlteredObject(ObjectType::Cluster),
+                        )),
+                        AlterClusterStrategyCondition::For(duration) => Ok(StageResult::Handle(
```

Comment:
This needs to be changed to a block that looks like https://github.com/MaterializeInc/materialize/blob/e0073189ce3d25881b1355cfe82170a5ff3a6cdf/src/adapter/src/coord/sequencer/inner/peek.rs#L435 including the `.instrument(span)` after the async block.